### PR TITLE
fix: add lifecycle hook to update carousel activeId when props change

### LINKED ISSUE #1828: There's no way to update activeId prop in Carousel
**Describe the bug; what happened?**
Carousel currently has a handled prop named activeId. This prop is mean to be used to programmatically change the current active slide. Unfortunately the logic that updates the state based on activeId prop seems to be only running in the constructor. This means that if I wanted to implement some outside mechanism to control updating the active slide, it just wouldn't work.

**What are the steps to reproduce the issue?**

Take a look at where the active id is updated in state:
https://github.com/microsoft/fast-dna/blob/master/packages/fast-components-react-msft/src/carousel/carousel.tsx

Basically it's done only in the constructor of the component - so no matter how many times I try to 're-render' with new props and new activeId, it will not work.

**What behavior did you expect?**
activeId should be taken into account every time I pass in via new props.

**If applicable, provide screenshots:**

**In what environment did you see the issue?**

* OS & Device: [e.g. MacOS, iOS, Windows, Linux] on [iPhone 7, PC]
* Browser [e.g. Edge, Chrome, Safari, FireFox]
* Version [e.g. 1.8.0]

**Is there any additional context?**

### PR DESCRIPTION
**Pull request by chrisdholt**
closes #1828 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ ] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-react-msft/src/carousel/carousel.tsx [hunk 1]
```diff
@@ -30,6 +30,22 @@ class Carousel extends Foundation<
...
+        nextProps: CarouselProps,
+        prevState: CarouselState
+    ): null | CarouselState {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
```
Comment (prabashn) [inline packages/fast-components-react-msft/src/carousel/carousel.tsx]: is it ok to return empty state?

File: packages/fast-components-react-msft/src/carousel/carousel.tsx [hunk 2]
```diff
@@ -30,6 +30,22 @@ class Carousel extends Foundation<
 > {
     public static displayName: string = `${DisplayNamePrefix}Carousel`;
 
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
```
Comment (prabashn) [inline packages/fast-components-react-msft/src/carousel/carousel.tsx]: UTs?

Comment (chrisdholt) [reply packages/fast-components-react-msft/src/carousel/carousel.tsx]: added

File: packages/fast-components-react-msft/src/carousel/carousel.tsx (hunk 1 above)
Comment (chrisdholt) [reply packages/fast-components-react-msft/src/carousel/carousel.tsx]: This is based on the requirements of the lifecycle hook - either return a state object or null to not update. :)

**Pull request merged**